crypto/fipsmodule/rand/entropy/internal.h

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#ifndef OPENSSL_HEADER_CRYPTO_RAND_ENTROPY_INTERNAL_H
+#define OPENSSL_HEADER_CRYPTO_RAND_ENTROPY_INTERNAL_H
+
+#include <openssl/ctrdrbg.h>
+
+#include "../new_rand_internal.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+// I could make these array types!
+struct entropy_source {
+  int is_initialized;
+  int (*initialize)(void);
+  int (*cleanup)(void);
+  int (*seed)(uint8_t seed[CTR_DRBG_ENTROPY_LEN]);
+  int (*personalization_string)(uint8_t personalization_string[CTR_DRBG_ENTROPY_LEN]);
+  int (*prediction_resistance)(uint8_t pred_resistance[RAND_PRED_RESISTANCE_LEN]);
+  int (*randomize)(void);
+};
```

Comment:
At first I was going to push for everything should be handled in the PR that adds it, but I see this isn't going to main so I'll allow it since I imagine some of this will be changing over time.